### ISSUE TITLE
[A11Y] An open ActionList menu does not close when Tab is pressed

### ISSUE THREAD

**Issue by danatemple** [dequelabs/cauldron#2181]
When an ActionList is opened, so the ActionListItems are shown, and you press Tab, focus moves to the next focusable element on the page and the ActionList remains open.

### Short Description

I think the above was short enough.

### Impact

I would say this is Critical as an app using the component will fail accessibility testing (in my limited experience).

- [ ] Blocker
- [ X] Critical
- [ ] Serious
- [ ] Moderate
- [ ] Minor

### Steps To Reproduce:

This can be seen on the docs page here: https://cauldron.dequelabs.com/components/ActionMenu

### Current Behavior:

See above

### Expected Behavior:

I expect it should behave like the HTML native <select> element, and close when Tab is pressed.
Focus should remain on the button element, not move to the next element.

See https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/select
